.github/workflows/infrastructure.yml

```diff
@@ -0,0 +1,171 @@
+
+jobs:
+
+# 1. Setup infrastructure variables
+  setup:
+    name: Setup 🔧
+    runs-on: [ self-hosted, APIM, infrastructure ]
+    outputs:
+      environment: ${{ env.environment }}
+      timezone: ${{ env.timezone }}
+    steps:
+    - name: Environment 🧪
+      run: echo "Environment set to ${{ env.environment }}"
+
+    - name: Timezone 🌐
+      run: echo "Timezone set to ${{ env.timezone }}"
+
+# 2. Base infrastructure creation
+  base:
+    name: Base 🧱
+    needs: setup
+    environment: ${{ needs.setup.outputs.environment }}
+    runs-on: [ self-hosted, APIM, infrastructure ]
+    steps:
+
+    - name: Pre-production 💫
+      if: contains('["dev", "staging"]', env.target)
+      run: echo "type=Preproduction" >> $GITHUB_ENV
+
+    - name: Production 💫
+      if: ${{ 'prod' == env.target }}
+      run: echo "type=Production" >> $GITHUB_ENV
+
+    - name: Tags 🏷️
+      run: echo tags='Environment=${{ env.type }}' \
+        'Product=${{ env.product }}' \
+        'Team=development' >> $GITHUB_ENV
+
+    - name: Login 🔐
+      uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: Azure defaults ✨
+      uses: Azure/cli@1.0.4
+      with:
+        inlineScript: |
+          # Basic
+          az configure --defaults location=${{ vars.REGION }}
+          az configure --defaults group=rg-${{ env.product }}-${{ env.target }}-001
+
+          # AZ CLI extensions upgrade
+          az extension add --name containerapp --upgrade
+
+    - name: Resource group 🏗️
+      uses: Azure/cli@1.0.4
+      with:
+        inlineScript: |
+          az group create \
+          --name rg-${{ env.product }}-${{ env.target }}-001 \
+          --tags ${{ env.tags }}
+
+    
+    - name: Container registry 📦️
+      uses: Azure/cli@1.0.4
+      with:
+        inlineScript: |
+          az acr create \
+          --name cr${{ env.product }}${{ env.target }}001 \
+          --sku Standard \
+          --admin-enabled true \
+          --tags ${{ env.tags }}
+
+    - name: Virtual network 🧵
+      uses: Azure/cli@1.0.4
+      with:
+        inlineScript: |
+          az network vnet create \
+          --name vnet-${{ env.product }}-001 \
+          --address-prefix ${{ secrets.VNET_ADDRESS_PREFIX }} \
+          --subnet-name snet-${{ env.product }}-001 \
+          --subnet-prefixes ${{ secrets.VNET_SUBNET_PREFIX }} \
+          --tags ${{ env.tags }}
+          
+    # - name: VNET peer 🔀
+    #   uses: Azure/cli@1.0.4
+    #   with:
+    #     inlineScript: |
+    #       az network vnet peering create \
+    #       --name vnet-peer-${{ env.product }}-001 \
+    #       --vnet-name vnet-${{ env.product }}-001 \
+    #       --remote-vnet $(az network vnet show --subscription ${{ secrets.REMOTE_VNET_SUBSCRIPTION }} --resource-group ${{ secrets.VNET_REMOTE_RESOURCE_GROUP }} --name ${{ secrets.REMOTE_VNET }} --query 'id' -o tsv) \
+    #       --allow-forwarded-traffic 1 \
+    #       --allow-vnet-access 1
```

Comment:
Should we delete this?

Reply:
I guess this is left because of your comment
> Azure VNET peering has been disabled from IaC untill cross-subscription RBAC has been created.

Maybe we should raise a ticket for this future change, write this code as a potential implementation on the ticket, and remove this comment
Or if you want to keep the comment, we could write a TODO above with the ticket number